pacman/operations/rig_algorithms/utils.py

```diff
@@ -0,0 +1,93 @@
+"""Utility functions which may be of value to router implementations.
+
+Based on
+https://github.com/project-rig/rig/blob/master/rig/place_and_route/route/utils.py
+"""
+
+
+from pacman.operations.rig_algorithms.links import Links
+
+
+def longest_dimension_first(vector, start=(0, 0), width=None, height=None):
+    """List the (x, y) steps on a longest-dimension first route.
+
+    Parameters
+    ----------
+    vector : (x, y, z)
+        The vector which the path should cover.
+    start : (x, y)
+        The coordinates from which the path should start (note this is a 2D
+        coordinate).
+    width : int or None
+        The width of the topology beyond which we wrap around (0 <= x < width).
+        If None, no wrapping on the X axis will occur.
+    height : int or None
+        The height of the topology beyond which we wrap around (0 <= y <
+        height).  If None, no wrapping on the Y axis will occur.
+
+    Returns
+    -------
+    [(:py:class:`~rig.links.Links`, (x, y)), ...]
+        Produces (in order) a (direction, (x, y)) pair for every hop along the
+        longest dimension first route. The direction gives the direction to
+        travel in from the previous step to reach the current step.
+        The first generated value is that of the first hop
+        after the starting position, the last generated value is the
+        destination position.
+    """
+    x, y = start
+
+    out = []
+
+    for dimension, magnitude in sorted(enumerate(vector),
```

Comment:
realign this. as it reads the sorted params should be on another line.

Reply:
yep looks better fixed.